### PR TITLE
Run lint fix on latest dependency update pull requests

### LINKED ISSUE #1639: Specify filepath for lint checks

### PR DESCRIPTION
**Pull request by gsheni**
- Fixes #1639

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,14 +9,14 @@ clean:
 .PHONY: lint
 lint:
 	python docs/notebook_version_standardizer.py check-execution
-	black . --config=./pyproject.toml --check
-	ruff . --config=./pyproject.toml
+	black woodwork/ docs/source/ --config=./pyproject.toml --check
+	ruff woodwork/ docs/source/ --config=./pyproject.toml
```
Comment (gsheni) [inline Makefile:13]:
```suggestion
	black . --config=./pyproject.toml --check
	ruff . --config=./pyproject.toml
```

File: Makefile [hunk 2]
```diff
@@ -15,8 +15,8 @@ lint:
 .PHONY: lint-fix
 lint-fix:
 	python docs/notebook_version_standardizer.py standardize
-	black . --config=./pyproject.toml
-	ruff . --config=./pyproject.toml --fix
+	black woodwork/ docs/source/ --config=./pyproject.toml
+	ruff woodwork/ docs/source/ --config=./pyproject.toml --fix
```
Comment (gsheni) [inline Makefile:19]:
```suggestion
	black . --config=./pyproject.toml
	ruff . --config=./pyproject.toml --fix
```

File: docs/source/release_notes.rst [hunk 3]
```diff
@@ -17,6 +17,7 @@ Future Release
         * Add codecov token for unit tests workflow (:pr:`1630`)
         * Add GitHub Actions cache to speed up workflows (:pr:`1631`)
         * Add pull request check for linked issues to CI workflow (:pr:`1633`, :pr:`1636`)
+        * Fix lint check on latest dependency update pull requests (:pr:`1640`)
```
Comment (gsheni) [inline docs/source/release_notes.rst:20]:
```suggestion
        * Run lint fix on latest dependency update pull requests (:pr:`1640`)
```

File: .github/workflows/latest_dependency_checker.yaml [hunk 4]
```diff
@@ -24,7 +24,6 @@ jobs:
       run: |
         python -m virtualenv venv_core
         source venv_core/bin/activate
-        python -m pip install --upgrade pip
```
Comment (bchen1116) [inline .github/workflows/latest_dependency_checker.yaml:27]: why are we removing this?

**Review by bchen1116 — COMMENTED** [review]
left one Q

File: .github/workflows/latest_dependency_checker.yaml (hunk 4 above)
Comment (gsheni) [reply .github/workflows/latest_dependency_checker.yaml:27]: It is run earlier:
https://github.com/alteryx/woodwork/blob/main/.github/workflows/latest_dependency_checker.yaml#L21

**Review by bchen1116 — APPROVED** [review]

**Pull request merged**